kraemer-lab/vneyard
# (Sampler_Brito) Add sampler

Add the Anderson Brito subsampler (https://github.com/andersonbrito/subsampler)

To test the module:
- Clone the originating repo (`git@github.com:jsbrittain/vneyard.git`) and checkout this branch (`git checkout sampler_brito`) [this will not be necessary once the module is added to the repository].
- Open GRAPEVNE and provide the repo base path (where you cloned `vneyard/`) as a local repository.
- Drag and drop `Subsampler_Brito` into the scene from the `EPI` project. Link a local folder as input using `LinkLocalFolder` (from the `Utility` project). You can use the sample data provided in `vneyard/workflows/EPI/modules/Subsampler_Brito/results/data`; do the same with the `config` folder (note that if you remove the subsampler config filenames from the module then it will run with defaults).
- `Build and Run` the workflow with the existing default options (or modify and build).
- `Open Results` to see the output; see below for a screenshot of the workflow with results.

Screenshot:
![Screenshot 2024-07-25 at 10 50 13](https://github.com/user-attachments/assets/43365b7c-4f1c-40fe-92fc-dda111099ad6)

resolves #2 

---

Improvements that can be made:
- [x] Make config input optional (i.e. if no config files found, use some defaults)
- [ ] Docstring can likely be improved with domain-specific knowledge
